### PR TITLE
Aggregate

### PR DESCRIPTION
**Pull request by ahmedadel56**
- Add a column species of type string to your animal's table. Modify your schema.sql file.

- [x] - Insert the following data:

- Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
- Animal: Her name is Plantmon. She was born on Nov 15th, 2022, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
- Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to 3 times.
- Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
- Animal: His name is Boarman. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
- Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
- Remember to add these insertions to your data.sql file.
- Inside a transaction update the animal's table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that species columns went back to the state before the transaction.

- [x] - Inside a transaction:

- Update the animal's table by setting the species column to Digimon for all animals that have a name ending on mon.
- Update the animal's table by setting the species column to pokemon for all animals that don't have species already set.
- Commit the transaction.
- Verify that change was made and persists after commit.
- Now, take a deep breath and... Inside a transaction delete all records in the animal's table, then roll back the transaction.
- After the rollback verify if all records in the animal's table still exist. After that, you can start breathing as usual ;)

- [x] - Inside a transaction:

- Delete all animals born after Jan 1st, 2022.
- Create a savepoint for the transaction.
- Update all animals' weight to be their weight multiplied by -1.
- Rollback to the savepoint
- Update all animals' weights that are negative to be their weight multiplied by -1.
- Commit transaction

- [x] - Write queries to answer the following questions:

- How many animals are there?
- How many animals have never tried to escape?
- What is the average weight of animals?
- Who escapes the most, neutered or not neutered animals?
- What is the minimum and maximum weight of each type of animal?
- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
- Remember to add these queries to your queries.sql file.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -5,4 +5,5 @@ CREATE TABLE animals (
     escape_attempts INT,
     neutered boolean,
     weight_kg DECIMAL,
+    species TEXT
```
Comment (oxenprogrammer) [inline schema.sql]: **Changes Required**
- [x] Let us not try and enter the column directly into `CREATE` table. You need to write a SQL command like `ALTER TABLE animals ADD species varchar(255);` to add the `species` column.

File: transaction.sql [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+SQL> UPDATE animals
+SET weight_kg = weight_kg * -1;
+-- Rollback to the savepoint
+SQL> ROLLBACK TO SP1;
+-- Update all animals' weights that are negative to be their weight multiplied by -1.
+SQL> UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0
+-- Commit transaction
+SQL> COMMIT;
```
Comment (oxenprogrammer) [inline transaction.sql]: **Changes Required**
- [x] Good effort, however it's unnecessary to create this file, i.e `transaction.sql`. The requirement dictates that we should do this inside `queries.sql`.

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]
Hi @ahmedadel56, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — COMMENTED** [review]
Hello @ahmedadel56 👋,

_This is an additional review as a result of the second opinion request._

- For the first comment request, please keep in mind that it's very important to practice all of the situations when you deal with databases. Such as altering an existing table! Therefore, it's one of the [requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Add%20a%20column%20species%20of%20type%20string%20to%20your%20animals%20table.%20Modify%20your%20schema.sql%20file.) to change the existing "animals" table by adding another column to it using the correct SQL command.
- For the second comment, I agree that you've implemented the queries correctly. However, to be as much specific and accurate as possible, it's dictated for the students to put all of the transactions in the `queries.sql` file. Please check the [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Remember%20to%20add%20these%20queries%20to%20your%20queries.sql%20file.).

Please make sure to apply the changes before you continue 😄

Keep rocking 🚀

File: schema.sql (hunk 1 above)
Comment (ahmedadel56) [reply schema.sql]: done

File: transaction.sql (hunk 2 above)
Comment (ahmedadel56) [reply transaction.sql]: done

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🎉✔**
Hello @ahmedadel56 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**